html/janus.js

```diff
@@ -2384,26 +2387,35 @@ function Janus(gatewayCallbacks) {
 						pluginHandle.consentDialog(true);
 					}
 					let constraints = Janus.trackConstraints(track), stream = null;
-					if(track.type === 'audio' || track.type === 'video') {
-						// Use getUserMedia: check if we need to group audio and video together
+					if(track.type === 'audio' || (track.type === 'video' || track.type === 'screen') ) {
+						// Set if getUserMedia should be used or if it should be getDisplayMedia
+						let captureUserMedia = track.type !== 'screen';
+						// Check if we need to group audio and video together
 						if(track.gumGroup) {
 							let otherType = (track.type === 'audio' ? 'video' : 'audio');
 							if(groups[track.gumGroup] && groups[track.gumGroup][otherType]) {
 								let otherTrack = groups[track.gumGroup][otherType];
+								if(!otherTrack && otherType === 'video') {
+									// If there other track is not video, then it is screen and should be a getDisplayMedia capture
```

Comment:
I'm trying to wrap up my mind around this check and the action it implements as a consequence, but I'm struggling to understand the logic behind it. Could you elaborate on why this should work that way? The check and the comment seem at odds.

Reply:
Sorry for my own late reply, been otherwise occupied.

Logic is a bit strange, yes. This is meant to address the issue of asking for audio before screen in the group.
If the audio track is handled first, we need to know what the other track type is. If it fails to get the other track and we had the type at video (from the other type ternary above) then it means the other type is actually screen and we should treat this as a display media request.